projects/demo/src/modules/components/input-month/examples/5/index.ts

```diff
@@ -1,18 +1,35 @@
-import {Component} from '@angular/core';
+import {Component, inject, LOCALE_ID} from '@angular/core';
 import {FormsModule} from '@angular/forms';
 import {changeDetection} from '@demo/emulate/change-detection';
 import {encapsulation} from '@demo/emulate/encapsulation';
-import type {TuiMonth} from '@taiga-ui/cdk';
+import {TuiMonth} from '@taiga-ui/cdk';
 import {TuiTextfield} from '@taiga-ui/core';
-import {TuiInputMonth} from '@taiga-ui/kit';
+import {TUI_MONTH_FORMATTER, TuiInputMonth} from '@taiga-ui/kit';
+import {of} from 'rxjs';
 
 @Component({
     standalone: true,
     imports: [FormsModule, TuiInputMonth, TuiTextfield],
     templateUrl: './index.html',
     encapsulation,
     changeDetection,
+    providers: [
+        {
+            provide: TUI_MONTH_FORMATTER,
+            useFactory: () => {
+                const formatter = Intl.DateTimeFormat(inject(LOCALE_ID), {
+                    year: '2-digit',
+                    month: 'short',
+                });
+
+                return of(
+                    (x: TuiMonth | null) =>
+                        (x && formatter.format(x.toLocalNativeDate())) ?? '',
+                );
+            },
+        },
+    ],
```

Comment:
what if

```ts
    providers: [
        tuiMonthFormatterProvider(({locale}) =>
            of(
                (x: TuiMonth | null) =>
                    (x &&
                        Intl.DateTimeFormat(locale, {
                            year: '2-digit',
                            month: 'short',
                        }).format(x.toLocalNativeDate())) ??
                    '',
            ),
        ),
    ],
```

Reply:
1. No need to recreate `Intl.DateTimeFormat` on every callback execution
2. `inject(LOCALE_ID)` is not the only way to get locale (application can use its own custom token or etc)

I think no need to create additional entity `tuiMonthFormatterProvider` (at least, for now)